extensions/parquet/table/src/test/java/io/deephaven/parquet/table/ParquetTableReadWriteTest.java

```diff
@@ -468,6 +474,361 @@ private static void writeReadTableTest(final Table table, final File dest,
+        final Table fromDiskWithMetadata = readTable(commonMetadata);
+        assertTableEquals(inputData.sort("symbol", "epic_collection_id"),
+                fromDiskWithMetadata.sort("symbol", "epic_collection_id"));
+    }
+
+    @Test
+    public void testAllPartitioningColumnTypes() {
+        final TableDefinition definition = TableDefinition.of(
+                ColumnDefinition.ofString("PC1").withPartitioning(),
+                ColumnDefinition.ofBoolean("PC2").withPartitioning(),
+                ColumnDefinition.ofChar("PC3").withPartitioning(),
+                ColumnDefinition.ofByte("PC4").withPartitioning(),
+                ColumnDefinition.ofShort("PC5").withPartitioning(),
+                ColumnDefinition.ofInt("PC6").withPartitioning(),
+                ColumnDefinition.ofLong("PC7").withPartitioning(),
+                ColumnDefinition.ofFloat("PC8").withPartitioning(),
+                ColumnDefinition.ofDouble("PC9").withPartitioning(),
+                ColumnDefinition.of("PC10", Type.find(BigInteger.class)).withPartitioning(),
+                ColumnDefinition.of("PC11", Type.find(BigDecimal.class)).withPartitioning(),
+                ColumnDefinition.of("PC12", Type.find(Instant.class)).withPartitioning(),
+                ColumnDefinition.of("PC13", Type.find(LocalDate.class)).withPartitioning(),
+                ColumnDefinition.of("PC14", Type.find(LocalTime.class)).withPartitioning(),
+                ColumnDefinition.ofInt("data"));
+
+        final Table inputData = ((QueryTable) TableTools.emptyTable(10).updateView("PC1 =  (ii%2 == 0) ? `AA` : `BB`",
+                "PC2 = (ii % 2 == 0)",
+                "PC3 = (char)(65 + (ii % 2))",
+                "PC4 = (byte)(ii % 2)",
+                "PC5 = (short)(ii % 2)",
+                "PC6 = (int)(ii%3)",
+                "PC7 = (long)(ii%2)",
+                "PC8 = (float)(ii % 2)",
+                "PC9 = (double)(ii % 2)",
+                "PC10 = java.math.BigInteger.valueOf(ii)",
+                "PC11 = java.math.BigDecimal.valueOf((double)ii)",
+                "PC12 = java.time.Instant.ofEpochSecond(ii)",
+                "PC13 = java.time.LocalDate.ofEpochDay(ii)",
+                "PC14 = java.time.LocalTime.of(i%24, i%60, (i+10)%60)",
+                "data = (int)(ii)"))
+                .withDefinitionUnsafe(definition);
+
+        final File parentDir = new File(rootFile, "testAllPartitioningColumnTypes");
+        final ParquetInstructions writeInstructions = ParquetInstructions.builder()
+                .setMetadataRootDir(parentDir.getAbsolutePath())
+                .build();
+        writeKeyValuePartitionedTable(inputData, parentDir, "data", writeInstructions);
+
+        // Verify that we can read the partition values, but types like LocalDate or LocalTime will be read as strings
+        // Therefore, we cannot compare the tables directly
+        final Table fromDisk = readKeyValuePartitionedTable(parentDir, EMPTY).select();
+
```

Comment:
Comparing fromDisk with input data leads to this error
```
Expected type of PC4 is byte actual type is int
Expected type of PC5 is short actual type is int
Expected type of PC7 is long actual type is int
Expected type of PC8 is float actual type is double
Expected type of PC10 is class java.math.BigInteger actual type is int
Expected type of PC11 is class java.math.BigDecimal actual type is double
Expected type of PC13 is class java.time.LocalDate actual type is class java.lang.String
Expected type of PC14 is class java.time.LocalTime actual type is class java.lang.String
```

Basically types of columns are not deduced properly, unless we read from the metadata file. Then it works as expected, as shown in the next few statements.